xil_carma_basic_lanefollow_scenario/docker-compose.yml

```diff
@@ -0,0 +1,245 @@
+
+services:
+
+  carma-simulation:
+    privileged: true
+    image: ${DOCKER_ORG}/cdasim:${DOCKER_TAG}
+    networks:
+      carma_sim_net:
+        ipv4_address: 172.2.0.2
+    container_name: carma-simulation
+    environment:
+      - DISPLAY=${DISPLAY}
+      - QT_X11_NO_MITSHM=1
+    volumes:
+      - /tmp/.X11-unix:/tmp/.X11-unix
+      - /opt/carma-simulation/logs:/opt/carma-simulation/logs
+      - /opt/carma-simulation/scenario_config.json:/opt/carma-simulation/scenario_config.json
+    deploy:
+      resources:
+        reservations:
+          devices:
+            - capabilities: [gpu]
+    command: bash -c "./mosaic.sh -s Town04"
+
+## carma_1 vehicle
+  roscore_1:
+    image: ${DOCKER_ORG}/carma-base:${DOCKER_TAG}
+    networks:
+      carma_net_1:
+        ipv4_address: 172.3.0.2
+    container_name: roscore_1
+    volumes_from:
+      - container:carma-config:ro
+    volumes:
+      - /opt/carma/.ros:/home/carma/.ros
+      - /opt/carma/logs/carma_1:/opt/carma/logs
+    restart: always
+    environment:
+      - ROS_IP=172.3.0.2
+      - ROS_MASTER_URI=http://172.3.0.2:11311/
+    command: roscore
+
+  platform_1:
+    image: ${DOCKER_ORG}/carma-platform:${DOCKER_TAG}
+    networks:
+      carma_net_1:
+        ipv4_address: 172.3.0.3
+    container_name: platform_1
+    volumes_from:
+      - container:carma-config:ro
+    volumes:
+      - /opt/carma/logs/carma_1:/opt/carma/logs
+      - /opt/carma/.ros:/home/carma/.ros
+      - /opt/carma/vehicle/calibration:/opt/carma/vehicle/calibration
+      - /opt/carma/yolo:/opt/carma/yolo
+      - /opt/carma/maps:/opt/carma/maps
+      - /opt/carma/routes:/opt/carma/routes
+      - /opt/carma/data:/opt/carma/data
+      - /opt/carma/simulation:/opt/carma/simulation
```

Comment:
Can we also add the configuration to access gpu from within this container or ros2 container? That seems to be required in order to run rviz

```
environment:
      - DISPLAY=${DISPLAY}
      - QT_X11_NO_MITSHM=1
    volumes:
      - /tmp/.X11-unix:/tmp/.X11-unix

```

Reply:
I added it to the platfor_ros2